src/layout/useLayout.ts

```diff
@@ -122,6 +104,11 @@ export const useLayout = ({
   const canvasHeight = pannable ? maxHeight : height;
   const canvasWidth = pannable ? maxWidth : width;
 
+  const scrollToXY = (xy: [number, number], animated = false) => {
+    ref.current.scrollTo({ left: xy[0], top: xy[1], behavior: animated ? 'smooth' : 'auto' });
+    setScrollXY(xy);
+  };
```

Comment:
I replaced the useEffect below 
```
  useEffect(() => {
    ref?.current?.scrollTo(scrollXY[0], scrollXY[1]);
  }, [scrollXY, ref]);
```

with this `scrollToXY` function

This is so we can control when or when not to animate a scroll. Animating the the initial scroll on render looks bad, but animating scrolls when fitting a node looks good

Reply:
I ended up passing an optional `animate` arg to alot of the functions that are used for scrolling and positioning too